armi/materials/material.py

```diff
@@ -32,6 +33,31 @@
 FAIL_ON_RANGE = False
 
 
+def parentAwareDensityRedirect(f):
+    """Wrap Material.density to warn people about potential problems.
+
+    If a Material is linked to a Component, ``Material.density`` may produce
+    different results from ``Component.density``. The component's density
+    is considered the source of truth because it incorporates changes in volume,
+    composition, and temperature in concert with the state of the reactor.
+    """
+
+    @functools.wraps(f)
+    def inner(self: "Material", *args, **kwargs) -> float:
```

Comment:
I've never seen this before.  You can type hint a variable as a string representing the class?

I guess I expected `self: Material`.

Reply:
Yeah it's a forward declaration. At this point in the file, `Material` is not defined. Wrapping type hints in strings is a way to type hint something that isn't known to the runtime interpreter at that time. Same approach can be used for wrapping things under the `typing.TYPE_CHECKING` import guard